search.py

```diff
@@ -37,13 +41,33 @@ def buildLink(self):
             self.searchQuery = "+".join(self.searchRaw)
         #end of search exceptions
         self.url = "http://www." + self.engine + "." + self.domain + self.searchString + self.searchQuery
-        return self.url
-    #end of link building
 
-    def openBrowser(self):
-        webbrowser.open_new_tab(self.url)
-    #end of openBrowser()
+        html = self.url; 
+        req = urllib.request.Request(html, headers={'User-Agent': 'Mozilla/5.0'})
+
+        soup = BeautifulSoup(urlopen(req).read(),"html.parser")
```

Comment:
I have a feeling this is part of another PR that you're working on.  I think this conversation should occur on another issue however I do like the direction you're taking it in. Going off of what I can read I think you're trying to scrape the links from the site to populate some results. This is great but I don't think it should be part of the base functionality. 

I think this should be put behind a command line argument for now so the users can choose between simply opening the link or having a few results pop up.

Regardless, this is a bit off topic for this PR. Let's just keep this particular PR about the test so the changes are easier to track.